### DISCUSSION TITLE
Please allow adding Optional System Fields to the created collection

### DISCUSSION THREAD

**Discussion by cdwmhcc** [directus/directus#4686] [Feature Requests]
Reason 1: I created a collection and added a lot of fields before discovering that I forgot to add fields such as `created_by` `updated_by` ...
Reason 2: I migrated data from v8. Many System Fields in v8 have changed and need to be added again.

There may be an integrated solution, but I didn't find it.

**Comment by benhaynes** [comment]
You can just create a datetime or UUID field and use the "on create" or "on update" values.

![Screen Shot 2021-03-26 at 9 25 06 AM](https://user-images.githubusercontent.com/522079/112638219-65e69600-8e15-11eb-958f-0fd05cc3e4a4.png)
![Screen Shot 2021-03-26 at 9 26 21 AM](https://user-images.githubusercontent.com/522079/112638225-67b05980-8e15-11eb-8df9-4d634ed218a3.png)

**Reply by cdwmhcc** [reply]
For example, status has default color settings in the field settings. If I create a new status, I need to set them one by one, which is very troublesome.

**Reply by benhaynes** [reply]
No way around that as of now. There could be an App/UI option to add those after the fact, but those helpers are still pretty opinionated, so we don't want to push them too hard in the platform.

**Reply by cdwmhcc** [reply]
No problem, this is not an important thing, just a suggestion.😘

**Comment by cborgia** [comment]
would be nice if the "optional fields" can be enabled AFTER creation without having to re-create
- as of now, appears as if we still would need to opt into at creation at the time of collection setup else as the reply above suggests, recreate when neded

**Reply by rijkvanzanten** [reply]
You can. The optional fields are nothing but presets for some commonly used fields in real life. There's nothing stopping you from creating those fields after the fact though 🙂

**Reply by cborgia** [reply]
thanks @rijkvanzanten for confirming the process is still to "recreate those fields after" ...yes, while the same end result can be achieved, one approach would take 2 seconds (check 6 checkboxes), the current approaches takes 3-5 minutes. nonetheless, love the tool and this is an easy work around, + once you make this mistake once you tend not to forgot next time...

**Reply by cf-ts** [reply]
Would be a nice feature anyways to generate fields from presets... Often  I'm doing the same settings all over and even if I copy some field, not all can be copied (i.e. relation fields) and not all settings are copied... so defining presets and/or  use system presets after creation really would be a time saver! :)

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by adgang** [comment]
I wrote a script to add system fields to an already defined collection:

Tested with:
Directus 10.6.x
[@directus/sdk](https://www.npmjs.com/package/@directus/sdk) 11.0.3



https://gist.github.com/adgang/caf5e0efacb393b1b8581640287c61df

**Comment by adgang** [comment]
This is a better script that will likely not be affected by directus version changes:

https://gist.github.com/adgang/1d83d1845476a291ff2106e6def0ef76
